src/apps/search-header/search-header.tsx

```diff
@@ -194,6 +196,7 @@ const SearchHeader: React.FC = () => {
   }
 
   function handleInputValueChange(changes: UseComboboxStateChange<Suggestion>) {
+    clearFilter();
```

Comment:
Please consider if we need to clear the filter at all given the addition to the search result in L163.

As far as I can tell this will clear the filter if no `usePersistedFilters` query parameter is set and no query parameter will be set by default.

This also means we keep the filter handler out of the search header, which I think is nice for separation of concerns.